File: test_archives/py_functions/batch4/function6.py
def function6(a, b, c, d, e):
    """
    This function takes in five arguments:
    a (string)
    b (integer)
    c (list)
    d (float)
    e (boolean)
   
    and returns the product of all the arguments.
    """
    product = a * b * d
    for x in c:
        product *= x
    if e:
        product *= 2
    return product

File: test_archives/cpp_functions/batch4/function6.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

double function6(string a, int b, vector<int> c, float d, bool e) {
    double product = a.size() * b * d;
    for (int x : c) {
        product *= x;
    }
    if (e) {
        product *= 2;
    }
    return product;
}